### DISCUSSION TITLE
blog/replenishment-optimization/

### DISCUSSION THREAD

**Discussion by giscus** [jeremy-feng/giscus#4] [Announcements]
# blog/replenishment-optimization/

https://fengchao.pro/blog/replenishment-optimization/

**Comment by guang9543** [comment]
生产率produce=pushub(i)/∑pushub（j）*销售均值。请问push ub(i）和 pushub（j）是用哪个数据计算？感谢解答（合掌）

**Reply by jeremy-feng** [reply]
$push\\\_ub_{i}$ 即 **push** **u**pper **b**ound of factory $i$，是指工厂 $i$ 的发货上限。

**Reply by guang9543** [reply]
感谢您的回答！（合掌）
还想问下您，图15 需求Demand Distribution的纵坐标demand是采用哪个数据呀？是对各DC各SKU历史销量数据的预测吗？

**Reply by guang9543** [reply]
另外，为什么使用历史销量的平均值来计算生产量呢？（可能有点啰嗦感谢解答！！）

**Reply by jeremy-feng** [reply]
Demand Distribution 中的 Demand 是各 (DC, SKU) 的历史销量的均值加上一个均匀分布（最多加减 0.1 倍历史销量的均值）的随机数作为噪音，具体的计算逻辑可以参考下面的代码：

![image](https://github.com/jeremy-feng/giscus/assets/44312563/9ff884df-bee5-4e0b-bc07-ab03fcfa0f27)

```python
import random
random.seed(0)

mean_sale = {
    (r["dc_id"], r["sku_id"]): r["mean_sale"] for _, r in df_mean_sale.iterrows()
}
for sku in sku_list:
    for dc in dc_list:
        if (dc, sku) not in mean_sale.keys():
            mean_sale[(dc, sku)] = 0


for key in mean_sale:
    noise = random.uniform(-0.1 * mean_sale[key], 0.1 * mean_sale[key])
    mean_sale[key] += noise

demand = mean_sale.copy()
```

**Reply by jeremy-feng** [reply]
> 另外，为什么使用历史销量的平均值来计算生产量呢？（可能有点啰嗦感谢解答！！）

使用历史销量的平均值是一种简单，但也有一定合理性的假设：历史上销量高，未来销量可能也高；历史上销量低，未来销量可能也低。未来销量高的 SKU，自然需要生产多一些。

当然，你也可以对销量预测进行建模。

**Reply by guang9543** [reply]
感谢您的耐心解答！真的非常感谢！祝愿您生活愉快、万事顺利！！

**Reply by jeremy-feng** [reply]
客气了，也谢谢你的关注！
